mcpserver/tools.go

```diff
@@ -218,71 +244,69 @@ var EnvironmentOpenTool = &Tool{
 	},
 }
 
-var EnvironmentCreateTool = &Tool{
-	Definition: mcp.NewTool("environment_create",
-		mcp.WithDescription(`Creates a new development environment.
-The environment is the result of a the setups commands on top of the base image.
-Read carefully the instructions to understand the environment.
-DO NOT manually install toolchains inside the environment, instead explicitly call environment_update`,
-		),
-		mcp.WithString("explanation",
-			mcp.Description("One sentence explanation for why this environment is being created."),
-		),
+type EnvironmentCreateToolRequest struct {
+	BaseRepositoryRequest
+
+	Title string `json:"title"`
+}
+
+var EnvironmentCreateTool = &Tool[EnvironmentCreateToolRequest, *environment.Environment]{
+	Definition: newRepositoryTool(
+		"environment_create",
+		`Creates a new development environment.
+	The environment is the result of a the setups commands on top of the base image.
+	Read carefully the instructions to understand the environment.
+	DO NOT manually install toolchains inside the environment, instead explicitly call environment_update`,
 		mcp.WithString("title",
 			mcp.Description("Short description of the work that is happening in this environment. Keep this title updated using `environment_update`."),
 			mcp.Required(),
 		),
-		mcp.WithString("environment_source",
-			mcp.Description("Absolute path to the source git repository for the environment."),
-			mcp.Required(),
-		),
 	),
-	Handler: func(ctx context.Context, request mcp.CallToolRequest) (*mcp.CallToolResult, error) {
-		repo, err := openRepository(ctx, request)
-		if err != nil {
-			return mcp.NewToolResultErrorFromErr("unable to open the repository", err), nil
-		}
-		title, err := request.RequireString("title")
+	ParseRequest: func(request mcp.CallToolRequest) (*EnvironmentCreateToolRequest, error) {
+		base, err := parseBaseRepositoryRequest(request)
```

Comment:
🐛 **Bug (High):** Need to properly extract and validate the required 'title' parameter before creating the request struct

```suggestion
		base, err := parseBaseRepositoryRequest(request)
		if err != nil {
			return nil, err
		}
		title, err := request.RequireString("title")
		if err != nil {
			return nil, err
		}
```